src/layers/domain/core/fhir_transform.py

```diff
@@ -0,0 +1,57 @@
+from typing import TypeVar
+
+from domain.core.product import ProductTeam
+from domain.core.root import Root
+from domain.fhir.r4 import cpm_model
+from domain.fhir.r4.models import Organization
+from domain.fhir.r4.strict_models import Organization as StrictOrganization
+from pydantic import BaseModel
+
+fhir_type = TypeVar("fhir_type")
+
+
+def create_product_team_from_fhir_json(fhir_json: dict, **kwargs) -> ProductTeam:
+    fhir_model = create_fhir_model_from_fhir_json(
+        fhir_json=fhir_json, fhir_models=[StrictOrganization], our_model=Organization
+    )
+    org = Root.create_ods_organisation(
+        id=fhir_model.partOf.identifier.id,
+        name=fhir_model.partOf.identifier.value,
+    )
+    user = Root.create_user(
+        id=fhir_model.contact[0].telecom[0].value, name=fhir_model.contact[0].name.text
+    )
+    (result, event) = org.create_product_team(fhir_model.id, fhir_model.name, user)
+    return result
```

Comment:
just to make a bit more human:

```suggestion
    (product_team, event) = org.create_product_team(fhir_model.id, fhir_model.name, user)
    return product_team
```